packages/odyssey-react-mui/src/Typography.tsx

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  Typography as MuiTypography,
+  TypographyProps as MuiTypographyProps,
+} from "@mui/material";
+import { ElementType, ReactNode } from "react";
+
+export const typographyVariantValues = {
+  h1: "h1",
+  h2: "h2",
+  h3: "h3",
+  h4: "h4",
+  h5: "h5",
+  h6: "h6",
+  body: "body1",
+  caption: "subtitle1",
+  legend: "legend",
+};
+
+export type TypographyProps = {
+  children: ReactNode;
+  classes?: object;
+  component?: ElementType;
+  variant?: keyof typeof typographyVariantValues;
+};
+
+export const Typography = ({
+  children,
+  classes,
+  component,
+  variant = "body",
+}: TypographyProps) => {
+  if (!component) {
+    component = variant;
+    if (variant === "body") {
+      component = "p";
+    }
+    if (variant === "caption") {
+      component = "h6";
+    }
+  }
```

Comment:
This is the result of some cleverness in MUI that makes things more difficult for us.

We want to allow consumers to change the `component` prop independently of the `variant`. The `component` prop, strangely, is _required_ in this implementation. It can't be undefined.

This conditional does the following:
- If `component` is explicitly defined, move on
- If not, set `component` to `variant`
- If `variant` is one of the two possibilities that doesn't map directly to an HTML element, set the `component` manually

It's ugly, it's brute-force, and it gets the job done.

Reply:
To throw another wrench in: we should probably alias the `body` variant to `<P>` for consistency, since `<body>` has a very different use. `<Caption>` is also a little 😬 because `<caption>` is a `<table>`-only element in HTML.

I'm in favor of this pattern, but it might start to feel limiting for some folks.